### PR TITLE
Increase table sticky header offset

### LINKED ISSUE #16960: Sticky table headers in app are not visible
### Describe the Bug

App table headers are not visible when scrolling over long lists.

### To Reproduce

https://user-images.githubusercontent.com/16063650/210034323-b6b7e0a9-fa91-4d6e-a822-49dd7ddf182b.mov

### Errors Shown

No

### What version of Directus are you using?

9.21.2

### What version of Node.js are you using?

16.18.1

### What database are you using?

SQLite

### What browser are you using?

Chrome

### How are you deploying Directus?

Directus Cloud

### Expected behavior

https://user-images.githubusercontent.com/16063650/210034346-67d4ba1f-5156-4b6a-89af-5b0b571e57f9.mov

### PR DESCRIPTION
**Pull request by Saissaken**
## Description

Sticky table headers are not visible while scrolling

Fixes #16960

## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

**Review by azrikahar — CHANGES_REQUESTED** [review]
Very nice catch! Upon further investigation, I think the root of the problem is that the _other_ `--v-table-sticky-offset-top` (not the global one modified in this PR):

https://github.com/directus/directus/blob/dc1a218c58110fc28ec58a43b817069ec72df173/app/src/layouts/tabular/tabular.vue#L308

is not inheriting the value of `--layout-offset-top`, as seen in your "Expected Behavior" screen capture:

chrome_5kn5XXeN9j

This is because although `--layout-offset-top` is defined here for `.layout` class:

https://github.com/directus/directus/blob/dc1a218c58110fc28ec58a43b817069ec72df173/app/src/modules/content/routes/collection.vue#L689-L691

and `.layout` class is being passed to the layout component:

https://github.com/directus/directus/blob/dc1a218c58110fc28ec58a43b817069ec72df173/app/src/modules/content/routes/collection.vue#L205

but ever since #7749 turned off attribute inheritance for all layouts, the `.layout` class is never really passed to the layout component anymore.

Here we can see when we remove `inheritAttrs: false` only without modifying anything else, the sticky header works again:

(yes there is a gap, but it is because `64px` is an outdated value, refer the solution mentioned below for more info)

https://user-images.githubusercontent.com/42867097/210040473-d760d402-1a8c-44fb-8910-8bbc3daf8da0.mp4

---

I believe the solution here is to add `:class="$attrs.class"` to all the layouts, for example here for tabular layout:

https://github.com/directus/directus/blob/dc1a218c58110fc28ec58a43b817069ec72df173/app/src/layouts/tabular/tabular.vue#L2

so that they will start to inherit the `.layout` class again.

Then because #7972 bumped down header-bar from `65px` to `61px`, we should also change all `--layout-offset-top: 64px;` (65px - 1px) to `--layout-offset-top: calc(var(--header-bar-height) - 1px);` instead, as seen in:

https://github.com/directus/directus/blob/dc1a218c58110fc28ec58a43b817069ec72df173/app/src/modules/settings/routes/roles/item/components/permissions-overview-header.vue#L23

**Comment by jay-p-b-7span** [comment]
> 

Very nice investigation. I have reviewed the issue and it looks like based on the current implementation, binding `$attrs.class` is a good and minimal solution as described by @azrikahar

**Comment by Saissaken** [comment]
Hey there! I been in vacations mode and almost forgot about this!

@azrikahar do you think this is good to go?

File: app/src/layouts/tabular/tabular.vue [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <template>
-	<div class="layout-tabular">
+	<div :class="$attrs.class" class="layout-tabular">
```
Comment (azrikahar) [inline app/src/layouts/tabular/tabular.vue:2]: let's make sure to add this to other layouts as well 👍 (so the Files module with Cards layout can also benefit from this for example)

**Review by azrikahar — CHANGES_REQUESTED** [review]
In hindsight, although I do still like the minimalistic aspect of my proposed solution, I guess it might also end up being some form of a "gotcha" in the sense that we'll need to remember to include `:class="$attrs.class"` in future layouts, including layout extensions 🤔

But the current solution should suffice (unless deemed otherwise)! 👍 I'm just thinking out loud to throw this thought out there.

**Comment by Nitwel** [comment]
@benhaynes I wonder what your thoughts are on making the table header sticky again?

TLDR: Header for `v-table` was once sticky, changes made it non sticky by mistake.

File: app/src/layouts/tabular/tabular.vue (hunk 1 above)
Comment (Nitwel) [reply app/src/layouts/tabular/tabular.vue:2]: I don't like this approach as this is unintuitive and easy to miss when writing new layouts.
I'd rather apply the `--layout-offset-top` not to the element with the layout class but the whole private view like so:
```css
.private-view {
    --layout-offset-top: calc(var(--header-bar-height) - 1px);
}
```

That way you always have access to `--layout-offset-top` no matter if you do `:class="$attrs.class"` or not. Also don't forget to remove `class="layout"` from the component as it has no purpose.

**Comment by benhaynes** [comment]
@Nitwel — I am a fan of the table component/layout sticky... although I would want to make sure there was no good/functional reason we removed it previously (CC @rijkvanzanten ). Also, if we add it back in, let's ensure the layout doesn't have a gap above it (I can see 1px of content scrolling above it in one of the videos here.

**Comment by rijkvanzanten** [comment]
I don't think the disabling was intentional 👍🏻

File: app/src/layouts/tabular/tabular.vue (hunk 1 above)
Comment (azrikahar) [reply app/src/layouts/tabular/tabular.vue:2]: > I don't like this approach as this is unintuitive and easy to miss when writing new layouts.

This is definitely something I'm worried about in my latest review! Whole hearted agree with your solution here 👍

**Comment by koksikus** [comment]
This looks much better.  This greatly improves the readability of the data.

**Comment by Nitwel** [comment]
I'll close this in favour of #17850 as this PR hasn't seen any activity in a month.
